internal/tcpchecker/logreporter.go

```diff
@@ -0,0 +1,46 @@
+package tcpchecker
+
+import (
+	"time"
+
+	"github.com/prometheus/common/log"
+)
+
+type LogReporter struct {
```

Comment:
Maybe we should move the reporter implementations into a `reporter` package instead? We can do that in another PR, just wanted to note it.

We now have a `CompositeReporter`, `LogReporter` and `PrometheusReporter`. That smells like a `reporter` package with `reporter.Composite`, `reporter.Log` and `reporter.Promtheus` that can then be configured as one expects.